node/pkg/driver/node_test.go

```diff
@@ -895,23 +896,129 @@ func TestNodeUnpublishVolume(t *testing.T) {
 }
 
 func TestNodeGetVolumeStats(t *testing.T) {
+	volId := "someStorageType:vol-test"
+	volumePath := "/test/path"
+	stagingTargetPath := "/staging/test/path"
+	mpathDevice := "/dev/dm-4"
+	volumePathWithHostPrefix := GetPodPath(volumePath)
+	mockCtl := gomock.NewController(t)
+	defer mockCtl.Finish()
+	mockNodeUtils := mocks.NewMockNodeUtilsInterface(mockCtl)
+	mockOsDeviceConHelper := mocks.NewMockOsDeviceConnectivityHelperScsiGenericInterface(mockCtl)
+	d := newTestNodeServiceExpand(mockNodeUtils, mockOsDeviceConHelper, nil)
+	req := &csi.NodeGetVolumeStatsRequest{
+		VolumeId:          volId,
+		VolumePath:        volumePath,
+		StagingTargetPath: stagingTargetPath,
+	}
 
-	req := &csi.NodeGetVolumeStatsRequest{}
+	testCases := []struct {
+		name     string
+		testFunc func(t *testing.T)
+	}{
+		{
+			name: "fail volumePath does not exists",
+			testFunc: func(t *testing.T) {
+				expErrCode := codes.NotFound
+				mockNodeUtils.EXPECT().GetPodPath(volumePath).Return(volumePathWithHostPrefix)
+				mockNodeUtils.EXPECT().IsPathExists(volumePathWithHostPrefix).Return(false)
 
-	d := newTestNodeService(nil, nil, nil)
+				_, err := d.NodeGetVolumeStats(context.TODO(), req)
+				assertError(t, err, expErrCode)
+			},
+		},
+		{
+			name: "fail to get stats",
+			testFunc: func(t *testing.T) {
+				expErrCode := codes.Internal
+				expSubString := "Failed to get statistics"
+				mockNodeUtils.EXPECT().GetPodPath(volumePath).Return(volumePathWithHostPrefix)
+				mockNodeUtils.EXPECT().IsPathExists(volumePathWithHostPrefix).Return(true)
+				mockNodeUtils.EXPECT().IsBlock(volumePathWithHostPrefix).Return(false, nil)
+				mockNodeUtils.EXPECT().GetFileSystemVolumeStats(volumePathWithHostPrefix).Return(driver.VolumeStatistics{}, status.Errorf(codes.Internal, "fail to get stats"))
```

Comment:
`status` is part of grpc and node utils should not return grpc errors

Reply:
done

Reply:
I don't see any change

Reply:
done now